src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedIntoPrisonEventService.kt

```diff
@@ -35,19 +42,25 @@ class PrisonerReceivedIntoPrisonEventService(
   private fun PrisonerReceivedAdditionalInformation.processPrisonerAdmissionEvent(eventOccurredAt: Instant) {
     log.info { "Processing Prisoner Admission Event for prisoner [$nomsNumber]" }
 
-    ciagKpiService?.processPrisonerAdmission(
-      prisonNumber = nomsNumber,
-      prisonAdmittedTo = prisonId,
-      eventDate = eventOccurredAt,
-    )
-
-    /*
-    TODO - replace the above call to the CiagKpiService with the following logic / calls to the InductionScheduleService and ReviewScheduleService
-       - If prisoner has no Induction, Action Plan or Induction Schedule, then create the Induction Schedule
-       - If prisoner has an Induction Schedule that is complete, create or update their Review Schedule with status SCHEDULED, reason PRISONER_READMISSION and due date +10 days
-       - if prisoner has an Induction Schedule, and it is not complete, reschedule it with the same deadline date rules as if it were a new admission (even if it was exempt (exempt thru transfers handled below))
-     (check all of the above rules with Aliki)
-     */
+    try {
+      // Attempt to create the prisoner's Induction Schedule
+      inductionScheduleService.createInductionSchedule(
+        prisonNumber = nomsNumber,
+        prisonerAdmissionDate = LocalDate.ofInstant(eventOccurredAt, ZoneOffset.UTC),
+      )
+    } catch (e: InductionScheduleAlreadyExistsException) {
+      // Prisoner already has an Induction Schedule
+      when (e.inductionSchedule.scheduleStatus) {
+        COMPLETED -> {
+          // The Induction was completed so we need to reschedule their active Review Schedule if they have one, or create a new Review Schedule.
+          rescheduleOrCreatePrisonersReviewSchedule(nomsNumber)
+        }
+        else -> {
+          // The Induction was not completed so need to reschedule it with a new deadline date.
+          // TODO - reschedule it
```

Comment:
Addressing this TODO should be the last thing we need to do for this ticket/refactoring, at which point I _think_ all of the handling of a prisoner admission in respect of their Induction Schedule and/or Review Schedule _should_ be done 🤞